cellpack/autopack/DBRecipeHandler.py

```diff
@@ -688,10 +722,20 @@ def _get_grad_and_obj(obj_data, obj_dict, grad_dict):
         obj_name = obj_data["name"]
         for key, target_dict in CompositionDoc.KEY_TO_DICT_MAPPING.items():
             if key in obj_data:
-                item_name = obj_data[key]["name"]
-                target_dict = grad_dict if key == "gradient" else obj_dict
-                target_dict[item_name] = obj_data[key]
-                obj_dict[obj_name][key] = item_name
+                # single gradient and inherited object
+                if isinstance(obj_data[key], dict):
+                    item_name = obj_data[key]["name"]
+                    target_dict = grad_dict if key == "gradient" else obj_dict
+                    target_dict[item_name] = obj_data[key]
+                    obj_dict[obj_name][key] = item_name
+                # combined gradients
+                elif key == "gradient" and isinstance(obj_data[key], list):
+                    new_grad_list = []
+                    for grad in obj_data[key]:
+                        for name in grad:
+                            grad_dict[name] = grad[name]
+                            new_grad_list.append(name)
+                    obj_dict[obj_name][key] = new_grad_list
```

Comment:
This section of code shares some similarities with `def get_reference_in_obj()` in `CompositionDoc` class, but they perform distinct tasks -- this one unpacks gradients for downloading, the latter resolves db references for duplicate checking while uploading. For clarity and easier maintenance, I'm going to leave them as standalone sections.